From 07ad14c7dbf592c1d30b7581a5da6d32a7123d96 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Thu, 15 Aug 2013 14:25:46 +0800
Subject: [PATCH 3/4] ENGR00275267 Settings: fix NPE in tethering setting

Fix one NUll Point Exception here when we not support BT feature.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 src/com/android/settings/TetherSettings.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/com/android/settings/TetherSettings.java b/src/com/android/settings/TetherSettings.java
index f0b6b44..4e0933d 100644
--- a/src/com/android/settings/TetherSettings.java
+++ b/src/com/android/settings/TetherSettings.java
@@ -387,6 +387,8 @@ public class TetherSettings extends SettingsPreferenceFragment
         }
 
         BluetoothAdapter adapter = BluetoothAdapter.getDefaultAdapter();
+        if (adapter == null)
+            return;
         int btState = adapter.getState();
         if (btState == BluetoothAdapter.STATE_TURNING_OFF) {
             mBluetoothTether.setEnabled(false);
-- 
1.8.0

